app/src/main/java/org/coepi/android/ui/notifications/NotificationShower.kt

```diff
@@ -27,20 +25,42 @@ class NotificationsShower(
     private val context: Context
 ) {
     private val channelId: String = "infection_contact_report"
-    private val notificationId: Int = 0
 
     fun showNotification(config: NotificationConfig) {
         with(NotificationManagerCompat.from(context)) {
-            notify(notificationId, notificationBuilder(config).build())
+            notify(config.notificationId, notificationBuilder(config).build())
         }
     }
 
-    private fun pendingIntent(args: NotificationIntentArgs): PendingIntent = PendingIntent.getActivity(
-        context, 0, Intent(context, MainActivity::class.java).apply {
-            flags = FLAG_ACTIVITY_SINGLE_TOP
-            putExtra(args.key.toString(), args.value)
-        }, FLAG_UPDATE_CURRENT
-    )
+    /**
+     * TODO: need a way to cancel specific system notifications using the unique notificationId that's
+     * apart of the [NotificationConfig]
+     */
+    fun cancelNotification(notificationId: Int) {
+        val notificationManager: NotificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.cancel(notificationId)
+    }
+
+    fun cancelAllNotifications() {
+        val notificationManager: NotificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.cancelAll()
+    }
+
+    fun getSystemNotificationList(): Array<out StatusBarNotification>? {
```

Comment:
This is exposing implementation details. `StatusBarNotification` is from `android.service.notification` and makes difficult to mock this class, e.g. for a Unit Test. Do you really need to return the list? From the current usage it seems you only need to know if we're showing the notifications, so you can expose instead something like `isShowingNotifications()` (which isn't entirely correct either, as we should check the channel id, as discussed, but ok for now if you'll fix this in a follow up PR)